### ISSUE TITLE
How do I build this?

### ISSUE THREAD

**Issue by JL2210** [jlagarespo/minecraft-cpp#1]
This looks interesting... How do I build it?

**Comment by jlagarespo** [comment]
You can just manually build it using a C++ compiler and then linking it with the proper libraries, however I think you could just use the IDE I use([codeblocks](http://www.codeblocks.org/)), put the proper settings there and let it build itself. I know there is no proper build instructions nor there is any dependencies list or anything like that to be nice and simple for people trying to build it, however it is just something random that I wanted to make, that is nothing playable yet, and will probably never be.

If you still want to build it anyway, the things you will need(which should work on both windows and linux) are:
 - [SFML](https://www.sfml-dev.org/)
 - [GLM](https://glm.g-truc.net/0.9.9/index.html)
 - OpenGL(of course)
 - [GLEW](http://glew.sourceforge.net/) (also known as modern opengl)
and you should be good to go.

Don't expect anything good or useful anyways, for the moment...

**Comment by jlagarespo** [comment]
Update: Now you use CMake to build it.

**Comment by JL2210** [comment]
Nice. I'll give it a shot.

**Comment by JL2210** [comment]
I did have to replace `ext.hpp` with `glm/ext.hpp` in `src/Minecraft.h`.

**Comment by jlagarespo** [comment]
Nice.
Currently working on textures
![imagen](https://user-images.githubusercontent.com/26935885/64911540-4b613600-d723-11e9-95a1-e9518001fce1.png)

**Comment by JL2210** [comment]
It's giving me some GLSL errors.

**Comment by jlagarespo** [comment]
Is it?
May I see the console output please?

**Comment by jlagarespo** [comment]
Could you please post the console output?

**Comment by JL2210** [comment]
Well, first of all, `block.h` is missing. I had to replace it with `Block.h`. Then I had to add `-I/usr/include/glm` to the compiler flags and change all instances of `mem.h` to `libavutil/mem.h`. After that, it compiled but didn't work. At that point, I just `rm -rf mcpp` and finished doing something else I was working on at the time. I can see what I can do about the log. Hang on, let me boot up my other computer...

**Comment by jlagarespo** [comment]
I just replaced ``block.h`` that for some reason my compiler toolchain found nicely.
About the GLSL errors, not sure what happened, either your drivers have some unsual GLSL implementation, or maybe in the push that I made the code was broken? Not sure.

Also, are you running on linux, or on the linux subsystem for windows (for rm -rf)? Because I have never played aroung with OpenGL or CMake in anything else than windows, so maybe I'll have to implement things a little differently for linux.

Either way, if it doesn't work this time around, don't bother to keep trying to make it work, as there really isn't much to see yet...

**Comment by JL2210** [comment]
I'm running on Linux, yes.

**Comment by JL2210** [comment]
Windows doesn't treat files with caps any different than files without caps. Linux does.

Also, on Linux, it's `-lGL` and `-lGLEW` instead of all the `-lopengl32` stuff.

**Comment by jlagarespo** [comment]
Yes, I agree, developing in Windows can be extremly annoying for things like that, but if I use it, is just because there is just such an extensive amount of code and anything graphics-related (games, drivers, etc.)
I use linux for most of everything else (programming related).

**Comment by JL2210** [comment]
I just happen to have a system running now that I can build this on. I'll give it a shot.

**Comment by JL2210** [comment]
I built it (I had to bump the minimum CMake version back from 3.14 to 3.13 to build on Debian) and then I tried to run it, but it said GLSL 3.30 is not supported.

**Comment by jlagarespo** [comment]
Hmmm, that's weird, in theory it should work for 330.
Are you running on a compatibility profile? You maybe should try removing the ``core`` from ``#version 330 core`` in every shader.

**Comment by jlagarespo** [comment]
Also, let me warn you that this project is kinda dead: I'm in the process of developing a proper game engine for it to run on, because it was becoming a bit unmaintainable, but maybe that I will never end either. I don't know, I guess I'll see.
